### PR TITLE
Merge implementations of Interval, Rectangle, and Brick

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

I needed different boundary conditions in each direction in 2D, but it was only supported in 3D. Also, only the 1D domain supported log/inverse grid point distributions. So I merged the three implementations into one.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by knelli2** [comment]
@nilsvu will rebase since this is something we want

**Comment by nilsvu** [comment]
@sxs-collaboration/spectre-core-devs could someone review this one? I rebased it and it's running through CI now.

**Comment by nilsdeppe** [comment]
I can review this

File: src/Domain/CoordinateMaps/Distribution.cpp [hunk 1]
```diff
@@ -50,4 +57,108 @@ Options::create_from_yaml<domain::CoordinateMaps::Distribution>::create<void>(
...
+//     SingularityPosition: 0.
+namespace domain::CoordinateMaps::detail {
+
+struct SingularityPositionImpl {
+  static constexpr Options::String help = {
+      "Position of coordinate singularity."};
+  struct SingularityPosition {
+    using type = double;
+    static constexpr Options::String help = {
+        "Position of coordinate singularity. "
```
Comment (nilsdeppe) [inline src/Domain/CoordinateMaps/Distribution.cpp:74]: Did you want this to be printed to terminal on one line or add the line breaks like in the code?

File: tests/Unit/Domain/Creators/Test_Brick.cpp [hunk 2]
```diff
@@ -101,286 +101,293 @@ void test_brick() {
   const std::vector<std::array<size_t, 3>> refinement_level{{{3, 2, 4}}};
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Creators/Test_Brick.cpp:101]: rename to `Test_Rectilinear.cpp`?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash

File: src/Domain/CoordinateMaps/Distribution.cpp (hunk 1 above)
Comment (nilsvu) [reply src/Domain/CoordinateMaps/Distribution.cpp:74]: Auto-linebreaks in the output are fine, I just don't want to bother re-flowing the line breaks in the C++ code.

File: tests/Unit/Domain/Creators/Test_Brick.cpp (hunk 2 above)
Comment (nilsvu) [reply tests/Unit/Domain/Creators/Test_Brick.cpp:101]: We still have `Test_Interval.cpp`, `Test_Rectangle.cpp`, and `Test_Brick.cpp` that haven't changed very much (on purpose, to make sure the domains work the same as before the refactor). Seems fine, no?

Comment (nilsdeppe) [reply tests/Unit/Domain/Creators/Test_Brick.cpp:101]: Ah okay, yea, that's fine then.

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**